soroban-env-host/src/storage.rs

```diff
@@ -422,8 +422,11 @@ impl Storage {
             ));
         }
 
-        let mut new_live_until =
-            host.with_ledger_info(|li| Ok(li.sequence_number.saturating_add(extend_to)))?;
+        let mut new_live_until = host.with_ledger_info(|li| {
+            li.sequence_number
+                .checked_add(extend_to)
+                .ok_or_else(|| host.err_arith_overflow())
```

Comment:
Same, should be an internal error (we verify that the value is sane above, so this would only occur due to mis-configuration/bug).

Reply:
The `extend_to` is also passed externally, see reply above.

Reply:
`extend_to` is guarded by the configuration, and the misconfiguration should be an internal error.

Reply:
Fixed.